src/templates/finops-hub/modules/storage.bicep

```diff
@@ -76,9 +90,181 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' = {
   tags: union(tags, contains(tagsByResource, 'Microsoft.Storage/storageAccounts') ? tagsByResource['Microsoft.Storage/storageAccounts'] : {})
   properties: {
     supportsHttpsTrafficOnly: true
+    allowSharedKeyAccess: true
     isHnsEnabled: true
     minimumTlsVersion: 'TLS1_2'
     allowBlobPublicAccess: false
+    publicNetworkAccess: 'Enabled'
+    networkAcls: {
+      bypass: 'AzureServices'
+      defaultAction: enablePublicAccess ? 'Allow' : 'Deny'
+    }
+  }
+}
+
+resource scriptStorageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' =  {
+  name: scriptStorageAccountName
+  location: location
+  sku: {
+    name: 'Standard_LRS' //sku
+  }
+  kind: 'StorageV2'// 'BlockBlobStorage'
+  tags: union(tags, contains(tagsByResource, 'Microsoft.Storage/storageAccounts') ? tagsByResource['Microsoft.Storage/storageAccounts'] : {})
+  properties: {
+    supportsHttpsTrafficOnly: true
+    allowSharedKeyAccess: true
+    isHnsEnabled: false
+    minimumTlsVersion: 'TLS1_2'
+    allowBlobPublicAccess: false
+    publicNetworkAccess: 'Enabled'
+    networkAcls: {
+      bypass: 'AzureServices'
+      defaultAction: 'Deny'
+      virtualNetworkRules: [
+        {
+          id: scriptSubnetId
+          action: 'Allow'
+        }
+      ]
+    }
+  }
+}
+
+resource blobPrivateDnsZone 'Microsoft.Network/privateDnsZones@2024-06-01' = {
+  name: 'privatelink.blob.${environment().suffixes.storage}'
```

Comment:
Are there any name length/uniqueness concerns we need to worry about for any of these new resources?

Reply:
No.  The storage account name itself is guaranteed to be globally unique by the storage service.  We're just overriding the externally available privatelink CNAME for resources within the vNet.